src/main/java/org/folio/search/integration/KafkaMessageListener.java

```diff
@@ -39,7 +43,30 @@ public void handleEvents(List<ResourceEventBody> events) {
       .map(KafkaMessageListener::asInstanceResource)
       .collect(toList());
 
-    indexService.indexResources(resources);
+    resources.stream()
+      .collect(groupingBy(ResourceEventBody::getTenant))
```

Comment:
I am not sure that splitting the batch into severals is a good idea, I am sure that there is an another way to retrieve tenant languages without creating FolioExecutionContexts

Reply:
I have modified this piece so that we only split events into batch for converting but indexing is still happens at once.